backend/capellacollab/settings/modelsources/t4c/crud.py

```diff
@@ -16,6 +16,18 @@ def get_all_t4c_instances(db: Session) -> list[DatabaseT4CInstance]:
     return db.execute(select(DatabaseT4CInstance)).scalars().all()
 
 
+def get_t4c_instances_by_version(version_id: int, db: Session):
+    return (
+        db.execute(
+            select(DatabaseT4CInstance).where(
+                DatabaseT4CInstance.version_id == version_id
+            )
+        )
+        .scalars()
+        .all()
+    )
+
+
 def get_t4c_instance(id_: int, db: Session) -> DatabaseT4CInstance:
     return db.execute(
         select(DatabaseT4CInstance).where(DatabaseT4CInstance.id == id_)
```

Comment:
https://peps.python.org/pep-0008/#descriptive-naming-styles
`single_trailing_underscore_: used by convention to avoid conflicts with Python keyword`

`id()` is a built-in function of Python. It's recommended in PEP0008 to not overwrite build-in Python functions.